DelvUI/Interface/GeneralElements/HideHudConfig.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using Dalamud.Plugin;
+using DelvUI.Config;
+using DelvUI.Config.Attributes;
+using DelvUI.Enums;
+using Dalamud.Game.ClientState.Actors.Types;
+
+namespace DelvUI.Interface.GeneralElements
+{
+    [Serializable]
+    [Section("Misc")]
+    [SubSection("Hide HUD", 0)]
+    public class HideHudConfig : PluginConfigObject
+    {
+
+        [Checkbox("Hide DelvUI outside of combat")]
+        [Order(5)]
+        public bool HideOutsideOfCombat = false;
+
+        [Checkbox("Enable Extended Config")]
+        [CollapseControl(10, 0)]
+        public bool EnableExtended = false;
+
+        /**
+         * MOVE THIS METHOD TO IT'S OWN CLASS
+         **/
+
+        public new static HideHudConfig DefaultConfig()
+        {
+            return new HideHudConfig();
+        }
+    }
+}
```

Comment:
I'm not sure if we really want a complicated logic for the whole thing. I think initially the idea is just a generic "hide hud out of combat" feature.

However, if you really wanted to have each element configured individually, think you should do it like this:
* All elements could have a "HideConfig" that can be changed to different modes.
* The HudManager / HideHelper would be able to tell for a HideConfig if it should be visible at the moment.

That makes the feature a bit more powerful because it'll result in all elements having an individual setting that the user can change, instead of hardcoding the logic one by one in the code.

Reply:
Just in case you weren't aware yet. There's an Github issue and Trello card on this where we can discuss it.
#388 
https://trello.com/c/3PqWjVER/144-feature-hide-delvui-hud-discussion